File: src/Servers/NginxServer/echo.js
function echo(r) {
    var body = '';
    var headers = r.headersIn;
    for (var name in headers) {
        body += name + ': ' + headers[name] + '\n';
    }
    r.return(200, body);
}

function cookie(r) {
    var body = '';
    var raw = r.headersIn['Cookie'];
    if (raw) {
        var pairs = raw.split(';');
        for (var i = 0; i < pairs.length; i++) {
            var trimmed = pairs[i].replace(/^\s+/, '');
            var eq = trimmed.indexOf('=');
            if (eq > 0) {
                body += trimmed.substring(0, eq) + '=' + trimmed.substring(eq + 1) + '\n';
            }
        }
    }
    r.return(200, body);
}

function handler(r) {
    if (r.method === 'POST') {
        r.return(200, r.requestText || '');
    } else {
        r.return(200, 'OK');
    }
}

export default { echo, cookie, handler };

File: src/Servers/HAProxyServer/echo.lua
core.register_service("echo", "http", function(applet)
    local body = ""
    local hdrs = applet.headers
    for name, values in pairs(hdrs) do
        for _, v in ipairs(values) do
            body = body .. name .. ": " .. v .. "\n"
        end
    end
    applet:set_status(200)
    applet:add_header("Content-Type", "text/plain")
    applet:add_header("Content-Length", tostring(#body))
    applet:start_response()
    applet:send(body)
end)

core.register_service("cookie", "http", function(applet)
    local body = ""
    local hdrs = applet.headers
    if hdrs["cookie"] then
        for _, raw in ipairs(hdrs["cookie"]) do
            for pair in raw:gmatch("[^;]+") do
                local trimmed = pair:match("^%s*(.*)")
                local eq = trimmed:find("=")
                if eq and eq > 1 then
                    body = body .. trimmed:sub(1, eq-1) .. "=" .. trimmed:sub(eq+1) .. "\n"
                end
            end
        end
    end
    applet:set_status(200)
    applet:add_header("Content-Type", "text/plain")
    applet:add_header("Content-Length", tostring(#body))
    applet:start_response()
    applet:send(body)
end)

core.register_service("echo_body", "http", function(applet)
    local body = applet:receive()
    if body == nil then body = "" end
    applet:set_status(200)
    applet:add_header("Content-Type", "text/plain")
    applet:add_header("Content-Length", tostring(#body))
    applet:start_response()
    applet:send(body)
end)
